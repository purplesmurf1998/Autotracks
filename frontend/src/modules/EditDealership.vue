<template>
  <CCard class="mx-4 mb-0">
    <CCardBody class="p-4">
      <CForm>
        <h1>Step 2: Register new dealership!</h1>
        <p class="text-muted">The dealership is the account everything will be tied to. </p>
        <CLabel class="mb-2">Dealership Name</CLabel>
        <CInput
          placeholder="Name"
          autocomplete="name"
          v-model="dealershipAccount.dealershipName"
          @keyup="$emit('update:dealershipAccount.dealershipName', dealershipAccount.dealershipName);"
        />
        <CLabel class="mb-2">Dealership Description</CLabel>
        <CTextarea
          placeholder="Description"
          rows="9"
          v-model="dealershipAccount.dealershipDescription"
          @keyup="$emit('update:dealershipAccount.dealershipDescription', dealershipAccount.dealershipDescription);"
        />
      </CForm>
    </CCardBody>
  </CCard>
</template>

<script>
export default {
  props: ['dealershipAccount']
}
</script>