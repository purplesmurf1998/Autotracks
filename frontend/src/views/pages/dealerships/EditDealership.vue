<template>
  <div>
    <CAlert show color="danger" v-if="errorMessage" class="mb-2">{{
      errorMessage
    }}</CAlert>
    <CForm @submit.prevent="submitForm">
      <CInput
        label="Dealership Name"
        :lazy="false"
        :value.sync="updatedDealership.name"
        placeholder="Dealership Name"
      />

      <CTextarea
        label="Dealership Description"
        :lazy="false"
        :value.sync="updatedDealership.description"
        placeholder="Dealership Description"
        rows="6"
      />
      <CRow class="justify-content-center">
        <CButton color="primary" type="submit" id="save-dealership-changes" :disabled="disableButtons">
          Save
        </CButton>
        <CButton
          class="ml-1"
          color="danger"
          :disabled="disableButtons"
          @click="setEditingDealership(false)"
        >
          Cancel
        </CButton>
      </CRow>
    </CForm>
  </div>
</template>

<script>
const axios = require("axios");

export default {
  props: ["dealership", "setEditingDealership"],
  data() {
    return {
      errorMessage: null,
      updatedDealership: {
        name: this.dealership.name,
        description: this.dealership.description,
      },
      disableButtons: false,
    };
  },
  methods: {
    submitForm() {
      const data = this.updatedDealership;
      axios({
        method: "PUT",
        url: `${this.$store.state.api}/dealerships/${this.dealership._id}`,
        headers: {
          Authorization: `Bearer ${this.$store.state.auth.token}`,
        },
        data,
      })
        .then((response) => {
          if (response.data.success) {
            this.$router.go();
          } else {
            console.log(response.data);
            this.showErrorMessage(response.data.message);
          }
        })
        .catch((err) => {
          console.log(err.data);
          this.showErrorMessage(
            "Invalid field inputs. Dealership must have a name."
          );
        });
    },
    showErrorMessage(message) {
      this.errorMessage = message;
      setTimeout(() => {
        this.errorMessage = null;
      }, 5000);
    },
  },
};
</script>