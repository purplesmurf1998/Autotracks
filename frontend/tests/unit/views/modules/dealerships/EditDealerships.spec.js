import {createLocalVue, shallowMount} from '@vue/test-utils'
import CoreuiVue from '@coreui/vue-pro'
import Vuex from 'vuex'

import EditDealership from './../../../../../src/modules/dealerships/EditDealership.vue'

const localVue = createLocalVue()
localVue.use(CoreuiVue)
localVue.use(Vuex)

const store = new Vuex.Store({
  state: {
    sidebarShow: 'responsive',
    sidebarMinimize: false,
    asideShow: false,
    darkMode: false
  }
})

describe('EditDealership.vue', () => {
  const wrapper = shallowMount(EditDealership, { 
    store, 
    localVue,
    dealershipAccount: {
      dealershipName: "testName",
      dealershipDescription: "testDescription"
    } 
  })
  it('has a name', () => {
    expect(EditDealership.name).toBe('EditDealership')
  })
  it('should render correct content', () => {
    expect(wrapper.find('h1').text()).toMatch('Step 2: Register new dealership!')
    expect(wrapper.find('p').text()).toMatch('The dealership is the account everything will be tied to.')
  })
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
