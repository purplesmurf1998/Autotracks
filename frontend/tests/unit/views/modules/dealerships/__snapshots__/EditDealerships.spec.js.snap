// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditDealership.vue renders correctly 1`] = `
<ccard-stub
  class="mx-4 mb-0"
>
  <ccardbody-stub
    class="p-4"
  >
    <cform-stub>
      <h1>
        Step 2: Register new dealership!
      </h1>
       
      <p
        class="text-muted"
      >
        The dealership is the account everything will be tied to. 
      </p>
       
      <clabel
        class="mb-2"
      >
        Dealership Name
      </clabel>
       
      <cinput-stub
        autocomplete="name"
        lazy="400"
        placeholder="Name"
        type="text"
      />
       
      <clabel
        class="mb-2"
      >
        Dealership Description
      </clabel>
       
      <ctextarea-stub
        lazy="400"
        placeholder="Description"
        rows="9"
      />
    </cform-stub>
  </ccardbody-stub>
</ccard-stub>
`;
